vowpalwabbit/reductions/epsilon_decay.cc

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "epsilon_decay.h"
+
+#include "config/options.h"
+#include "global_data.h"
+#include "label_type.h"
+#include "learner.h"
+#include "memory.h"
+#include "model_utils.h"
+#include "prediction_type.h"
+#include "vw.h"
+
+#include <algorithm>
+#include <cmath>
+#include <string>
+
+using namespace VW::config;
+using namespace VW::LEARNER;
+
+// namespace logger = VW::io::logger;
+
+namespace VW
+{
+namespace epsilon_decay
+{
+void epsilon_decay_score::update_bounds(float w, float r)
+{
+  update(w, r);
+
+  // update the lower bound
+  distributionally_robust::ScoredDual sd = this->chisq.recompute_duals();
+  _lower_bound = static_cast<float>(sd.first);
+}
+
+template <class ForwardIt>
+ForwardIt swap_models(ForwardIt first, ForwardIt n_first, ForwardIt end)
```

Comment:
Put things what are not exported from the .h in an anonymous namespace

Reply:
I added these functions to these functions to the .h instead